### PR TITLE
samples: add sample for creating a bucket with hierarhical namespace …

### PR DESCRIPTION
**Pull request by ddelgrosso1**
…enabled

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Comment by ddelgrosso1** [comment]
Part of the work for: https://github.com/googleapis/google-cloud-node/issues/5417

File: samples/createBucketWithHierarchicalNamespace.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const {Storage} = require('@google-cloud/storage');
+
+  // Creates a client
+  // The bucket in the sample below will be created in the project associated with this client.
+  // For more information, please see https://cloud.google.com/docs/authentication/production or https://googleapis.dev/nodejs/storage/latest/Storage.html
+  const storage = new Storage();
+
+  async function createBucketWithHierarchicalNamespace() {
+    const [bucket] = await storage.createBucket(bucketName, {
+      hierarchicalNamespace: true,
```
Comment (tritone) [inline samples/createBucketWithHierarchicalNamespace.js:42]: Doesn't creating an HNS bucket require UBLA to be set to true as well? Or does that happen by default somehow in Node?

Comment (ddelgrosso1) [reply samples/createBucketWithHierarchicalNamespace.js:45]: Added. Somehow slipped by me from the 'canonical' sample.

File: samples/createBucketWithHierarchicalNamespace.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  // The bucket in the sample below will be created in the project associated with this client.
+  // For more information, please see https://cloud.google.com/docs/authentication/production or https://googleapis.dev/nodejs/storage/latest/Storage.html
+  const storage = new Storage();
+
+  async function createBucketWithHierarchicalNamespace() {
+    const [bucket] = await storage.createBucket(bucketName, {
+      iamConfiguration: {
+        uniformBucketLevelAccess: true,
+      },
+      hierarchicalNamespace: true,
```
Comment (bajajneha27) [inline samples/createBucketWithHierarchicalNamespace.js:45]: In other languages, I have seen `hierarchicalNamespace` having another field `enabled`. Is that not the case with Node?

**Review by bajajneha27 — APPROVED** [review]

**Pull request merged**